packages/core/src/node/initRsbuild.ts

```diff
@@ -244,16 +242,72 @@ async function createInternalBuildConfig(
           .rule('css-virtual-module')
           .test(/\.rspress[\\/]runtime[\\/]virtual-global-styles/)
           .merge({ sideEffects: true });
+
+        if (isServer) {
+          chain.output.filename('main.cjs');
+          chain.optimization.minimize(false);
+        }
       },
     },
+    environments: {
+      web: {
+        source: {
+          entry: {
+            index: CLIENT_ENTRY,
+          },
+          alias: {
+            ...reactCSRAlias,
+          },
+          define: {
+            'process.env.__SSR__': JSON.stringify(false),
+          },
+        },
+        output: {
+          target: 'web',
+          overrideBrowserslist: webBrowserslist,
+          distPath: {
+            // `root` must be a relative path in Rsbuild
+            root: csrOutDir,
+          },
+        },
+      },
+      ...(enableSSG
+        ? {
+            node: {
+              source: {
+                entry: {
+                  index: SSR_ENTRY,
+                },
+                alias: {
+                  ...reactSSRAlias,
+                },
+                define: {
+                  'process.env.__SSR__': JSON.stringify(true),
+                },
+              },
+              performance: {
+                printFileSize: false,
+              },
+              output: {
+                target: 'node',
+                overrideBrowserslist: ssrBrowserslist,
+                distPath: {
+                  // `root` must be a relative path in Rsbuild
```

Comment:
Actually Rsbuild allows to set absolute path as root, see: https://rsbuild.dev/config/output/dist-path#root-directory

Reply:
just copy it from the old code